### PR TITLE
React Image Gallery Post

### PR DESCRIPTION
**Pull request by ajeetchaulagain**

### CODE REVIEW

File: src/posts/react-image-gallery.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
+---
+title: 'How to use React Image Gallery to build a beautiful and responsive image gallery'
+date: '2022-09-06'
```
Comment (geshan) [inline src/posts/react-image-gallery.md:3]:
```suggestion
date: '2022-11-23'
```

File: src/posts/react-image-gallery.md [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+tags: ['javascript', 'react', 'image-gallery']
+thumbnail: ../images/thumbnails/image-gallery-icon.png
+author: 'Ajeet Chaulagain'
+---
+
+import { QuickInfo } from "../components/utilities/quick-info.tsx"
+
+Suppose you are creating apps or web pages. In that case, you might need an image gallery to share the multiple images on our webpage with the best user interactions possible for the targeted audience. However, creating an image library from scratch that provides a great user experience for to end user requires quite an effort. In this tutorial, we will learn how to use different* APIs of the react-image-gallery* npm package to create a _beautiful and elegant image gallery_ for your need with minimal effort.
```
Comment (geshan) [inline src/posts/react-image-gallery.md:11]: use AIDA method for a better start.https://www.siegemedia.com/creation/aida-model#:~:text=The%20AIDA%20model%20is%20a,to%20convert%20readers%20into%20customers.

File: src/posts/react-image-gallery.md [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+_Screenshot of final application:_
```
Comment (geshan) [inline src/posts/react-image-gallery.md:13]: the intro is not connecting to it.

Generally:
Title: What are you going to learn/benefit.

* Intro: What is coming up
* Body: The meat
* Conclusion: What you just read


:)

File: src/posts/react-image-gallery.md [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+![Final application screenshot](../images/react-image-gallery/final-application-screenshot.png)
+
+Here is the final source code and demo of the app:
+
+- [View Source code on Github](https://github.com/ajeetchaulagain/react-image-gallery-demo)
+- [View project demo](https://react-image-gallery-demo.ajeetchaulagain.com/)
+
+## Why react-image-gallery ?
+
+React image gallery is one of the most popular and community-loved npm packages for providing react components to _build highly customizable_ image galleries. Its popularity among consumers is due to its support for a wide range of features out of the box. These features include _swipe gestures for mobile devices, navigation through thumbnails, responsive design, and much more_. Moreover, the react component provided by the package is highly customizable through different props that will help almost all your job get done as a need of gallery in your project. In this tutorial, we will demonstrate the most used props that will be handy for our project.
```
Comment (geshan) [inline src/posts/react-image-gallery.md:24]: back most popular with a link to npm trends like this one: https://npmtrends.com/react-gallery-swiper-vs-react-grid-gallery-vs-react-image-gallery-vs-react-image-lightbox-vs-react-image-slider-vs-react-images

Comment (geshan) [reply src/posts/react-image-gallery.md:24]: Link features to react image gallery feature page or feature on github page.

File: src/posts/react-image-gallery.md [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+
+Once the project is created, _cd_ to the newly created directory and start the project.
+
+```bash
+cd react-image-gallery-demo && npm start
+```
+
+Running the above command will open the new app in _localhost:3000_
+
+### Directory and imports cleanup
```
Comment (geshan) [inline src/posts/react-image-gallery.md:44]: ### Clean up to create an image gallery

File: src/posts/react-image-gallery.md [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+
+The react app setup through create-react-app will have all the working files placed in a single directory src. The initial directory structure is something like this:
```
Comment (geshan) [inline src/posts/react-image-gallery.md:46]:
```suggestion
The React app setup through create-react-app will have all the working files placed in a single directory src. The initial directory structure is something like this:
```

All `React` not `react to a situation` :)

File: src/posts/react-image-gallery.md [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+
+![Final image gallery](../images/react-image-gallery/final-application-screenshot.png)
+
+Yahoo! We made beautiful and responsive image galleries. You can view the demo for this app [here](https://github.com/ajeetchaulagain/react-image-gallery-demo)
+
+## Customizing React image gallery
+
+We made the beautiful image gallery without configuring any props and relying on the defaults. We can also provide custom value to different props supported by packages for customizing as per needs. The [documentation for react-image-gallery](https://github.com/xiaolin/react-image-gallery#user-content-props) package is decent, and I suggest checking it out for a wide range of possible configurations.
+
+For example, Suppose we want our gallery to be auto-played and don't want thumbnails to display. In that case, we can pass _autoPlay_ props to _true_ and _showThumbnails_ props to _false_ in an _ImageGallery_ component.
```
Comment (geshan) [inline src/posts/react-image-gallery.md:196]:
```suggestion
For example, you want our gallery to be auto-played and don't want thumbnails to display. In that case, we can pass _autoPlay_ props to _true_ and _showThumbnails_ props to _false_ in an _ImageGallery_ component.
```

File: src/posts/react-image-gallery.md [hunk 8]
```diff
@@ -1,33 +1,31 @@
 ---
 title: 'How to use React Image Gallery to build a beautiful and responsive image gallery'
```
Comment (geshan) [inline src/posts/react-image-gallery.md:2]: Some files are too big for the Web like 478 kb and 1.3 mb. Use photopea.com to make them smaller.

File: src/posts/react-image-gallery.md [hunk 9]
```diff
@@ -65,7 +63,7 @@ To achieve the above cleanup:
 
 ### Adding CSS reset file
 
-To avoid browser default stylings for different HTML elements, I would like to add a CSS reset code provided by https://meyerweb.com/eric/tools/css/reset/. To add the CSS reset code in the project, copy and paste the code provided by above url, to a new file called _css-reset.css_ in _styles_ directory. And reference that newly created file in _index.css_ which acts as our entry point for all the CSS:
+To avoid browser default stylings for different HTML elements, Let's add a CSS reset code provided by https://meyerweb.com/eric/tools/css/reset/. To add the CSS reset code to the project, copy and paste the above code to a new file called _css-reset.css_ in the _styles_ directory. And reference that newly created file in _index.css_, which acts as our entry point for all the CSS:
```
Comment (geshan) [inline src/posts/react-image-gallery.md:66]: You will add...

File: src/posts/react-image-gallery.md [hunk 10]
```diff
@@ -185,27 +185,29 @@ header {
...
+![Image gallery without thumbnail and auto play](../images/react-image-gallery/image-gallery-without-thumbnai-and-autoplay.jpg)
 
 ## Conclusion
 
-This article should have given you a basic idea of how we can leverage third-party modules like react-image-gallery to build an image gallery with minimal effort.
+This tutorial gives you a basic idea of leveraging third-party modules like react-image-gallery to build an image gallery with minimal effort. I hope you enjoyed this article about using the react-image-gallery package through this step-by-step guide.
 
 - [View Source code on Github](https://github.com/ajeetchaulagain/react-image-gallery-demo)
 - [View Project Demo](https://react-image-gallery-demo.ajeetchaulagain.com/)
+- [React Image Gallery NPM Package](https://www.npmjs.com/package/react-image-gallery)
```
Comment (geshan) [inline src/posts/react-image-gallery.md:212]: Minor fixes, then go live:).

File: src/posts/react-image-gallery.md (hunk 8 above)
Comment (ajeetchaulagain) [reply src/posts/react-image-gallery.md:2]: Already converted to jpg and reduced size !

File: src/posts/react-image-gallery.md (hunk 10 above)
Comment (ajeetchaulagain) [reply src/posts/react-image-gallery.md]: Huss dai.

**Pull request merged**